x-pack/plugins/apm/server/routes/mobile/get_mobile_detailed_statistics_by_field.ts

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ProcessorEvent } from '@kbn/observability-plugin/common';
+import {
+  kqlQuery,
+  rangeQuery,
+  termQuery,
+} from '@kbn/observability-plugin/server';
+import { keyBy } from 'lodash';
+import { getBucketSize } from '../../../common/utils/get_bucket_size';
+import { getOffsetInMs } from '../../../common/utils/get_offset_in_ms';
+import { APMEventClient } from '../../lib/helpers/create_es_client/create_apm_event_client';
+import { environmentQuery } from '../../../common/utils/environment_query';
+import {
+  SERVICE_NAME,
+  TRANSACTION_DURATION,
+} from '../../../common/es_fields/apm';
+import { getLatencyValue } from '../../lib/helpers/latency_aggregation_type';
+import { LatencyAggregationType } from '../../../common/latency_aggregation_types';
+import { offsetPreviousPeriodCoordinates } from '../../../common/utils/offset_previous_period_coordinate';
+import { Coordinate } from '../../../typings/timeseries';
+
+interface MobileDetailedStatistics {
+  fieldName: string;
+  latency: Coordinate[];
+  throughput: Coordinate[];
+}
+
+export interface MobileDetailedStatisticsResponse {
+  currentPeriod: Record<string, MobileDetailedStatistics>;
+  previousPeriod: Record<string, MobileDetailedStatistics>;
+}
+
+interface Props {
+  kuery: string;
+  apmEventClient: APMEventClient;
+  serviceName: string;
+  environment: string;
+  start: number;
+  end: number;
+  field: string;
+  fieldNames: string[];
+  offset?: string;
+}
+
+async function getMobileDetailedStatisticsByField({
+  environment,
+  kuery,
+  serviceName,
+  field,
+  fieldNames,
+  apmEventClient,
+  start,
+  end,
+  offset,
+}: Props) {
+  const { startWithOffset, endWithOffset } = getOffsetInMs({
+    start,
+    end,
+    offset,
+  });
+
+  const { intervalString } = getBucketSize({
+    start: startWithOffset,
+    end: endWithOffset,
+    minBucketSize: 60,
+  });
+
+  const response = await apmEventClient.search(
+    `get_mobile_detailed_statistics_by_${field}`,
+    {
+      apm: {
+        events: [ProcessorEvent.transaction],
```

Comment:
do we need to always query transaction events? are there scenarios where we can use transaction metrics, maybe even rolled up?